internal/stoppropaganda/customresolver/customgetip.go

```diff
@@ -0,0 +1,47 @@
+package customresolver
+
+import (
+	"context"
+	"net"
+)
+
+var getIPcachedResolver = &CustomResolver{
+	ParentResolver: net.DefaultResolver,
+}
+
+// Modified to use stoppropaganda's CustomResolver
+// so that it caches DNS records
+func CustomLookupIP(host string, helperIPBuf []net.IP) ([]net.IP, error) {
+	addrs, err := getIPcachedResolver.LookupIPAddr(context.Background(), host)
+	if err != nil {
+		return nil, err
+	}
+	ips := helperIPBuf[:0]
+	for _, ia := range addrs {
+		ips = append(ips, ia.IP)
+	}
+	return ips, nil
+}
+
+func GetIPs(host string, helperIPBuf []net.IP) (ips []net.IP, err error) {
+	addr := net.ParseIP(host)
+	if addr == nil {
+		ips, err := CustomLookupIP(host, helperIPBuf)
+		if err != nil {
+			return nil, err
+		}
+		for i := 0; i < len(ips); i++ {
+			ip := ips[i]
+			if ipv4 := ip.To4(); ipv4 == nil {
+				// Remove inplace trick
```

Comment:
Thank you for putting some comments, trying to learn go ;)

Reply:
Yeah, that's a neat trick